[BUG] Fix On The Fly issues with mesh layer

vcloarec opened the pull request:
Fix #29330 and #28827

PeterPetrik commented:
please create official PR to qgis/QGIS

PeterPetrik commented on src/core/mesh/qgstriangularmesh.cpp:
```diff
@@ -140,6 +151,7 @@ bool QgsTriangularMesh::update( QgsMesh *nativeMesh, const QgsCoordinateTransfor
         Q_UNUSED( cse )
         QgsDebugMsg( QStringLiteral( "Caught CRS exception %1" ).arg( cse.what() ) );
         mTriangularMesh.vertices[i] = vertex;
```
when there is invalid vertex and we will not use it , I was thiniking to assign vertex.x = NaN , and vertex.y = NaN. and instead of `QSet<int> invalidVertexIndexes.contains(id)` you would just check in `std::isnan(vertex.x)`

vcloarec committed:
fix #29330 and #28827

replace context.extent() with context.mapExtent() where is needed


.
don't add faces with invalid vertices
improvement
